### DISCUSSION TITLE
Why is AESARadar actionable but seems to have no actions?

### DISCUSSION THREAD

**Discussion by Quinticx** [dstl/Stone-Soup#829] [Q&A]
I am currently trying to mimic the sensor management example but with the AESARadar instead of the RadarRotatingBearingRange. I thought this would be possible since both are actionable sensors, however, when printing out their respective actions, RadarRotatingBearingRange gives 'dwell_centre' as the actionable property which in-turn results in a DwellActionsGenerator, while the AESARadar gives back an empty set. Why is this?

**Comment by sdhiscocks** [comment]
@Quinticx So the AESA Radar doesn't currently have any actionable properties, as they haven't been implemented yet. It's a bit more complicated with the AESA Radar, as it uses models to direct the sweep of the beam, so an action generator would need to be created to handle switching the transition model, and setting the parameters.

**Reply by Quinticx** [reply]
Okay thanks! I was messing with this last night by making the beam transition model an actionable property and created a beam transition model change action and generator. The generator just makes a new stationary beam at every point in the field of view and the change action switches to the new model after the end time. Would it make more sense to do a linear interpolation between the old stationary beam and the new stationary beam, or to use a beam sweep transition model to move the beam? I can put up a PR today or tomorrow for it.

**Reply by nperree-dstl** [reply]
This sounds great, look forward to seeing the PR! From what I can understand it sounds like it would make sense to use a beam sweep transition model to move the beam (not a radar expert), but if the end result is the same/similar then either way is probably fine.

**Comment by apiszcz** [comment]
If there is any time for a tutorial on the implementation, it may benefit
others..

**Comment by Quinticx** [comment]
Sorry for the delay, life decided to right after I posted that it wanted to be crazy. I've made some progress on this and will attach the pull request here this weekend.
